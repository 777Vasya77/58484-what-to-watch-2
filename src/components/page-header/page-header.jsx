import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {selectors} from "~/selectors/selectors";

const PageHeader = (props) => {
  const {isAuth, authUser} = props;
  return (
    <header className="page-header movie-card__head">
      <div className="logo">
        <a href="/" className="logo__link">
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </a>
      </div>

      <div className="user-block">
        {isAuth
          ? <div className="user-block__avatar">
            <img src={authUser.fullAvatarUrl} alt={authUser.name} width="63" height="63"/>
          </div>
          : `Sign In`}
      </div>
    </header>
  );
};

export {PageHeader};

PageHeader.propTypes = {
  isAuth: PropTypes.bool.isRequired,
  authUser: PropTypes.object
};

const mapStateToProps = (state) => ({
  isAuth: selectors.user.isAuth(state),
  authUser: selectors.user.getAuthUser(state),
});

export default connect(mapStateToProps)(PageHeader);
