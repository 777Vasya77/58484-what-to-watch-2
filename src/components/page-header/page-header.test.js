import React from 'react';
import renderer from 'react-test-renderer';
import {PageHeader} from '~/components/page-header/page-header';

jest.mock(`react-router-dom`, () => ({
  Link: `Link`,
}));

describe(`PageHeader component tests`, () => {
  it(`Component render correctly`, () => {
    const tree = renderer
        .create(
            <PageHeader isAuth={false}/>
        ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
