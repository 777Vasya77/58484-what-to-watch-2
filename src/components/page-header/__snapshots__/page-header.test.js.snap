// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader component tests Component render correctly 1`] = `
<header
  className="page-header movie-card__head"
>
  <div
    className="logo"
  >
    <Link
      className="logo__link"
      to="/"
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </Link>
  </div>
  <div
    className="user-block"
  >
    <Link
      to="/login"
    >
      Sign In
    </Link>
  </div>
</header>
`;
